src/pages/teamHome/teamJoin.tsx

```diff
@@ -1,346 +1,387 @@
+    const deadline = new Date(adData.teamDate);
+    if (deadline < today) {
+      // if the deadline has passed
+      setValid(true);
+    } else {
+      setValid(false);
+    }
+    teamList.forEach((doc: any) => {
+      if (doc.data().members.length < adData.maxSize) {
+        const allNames: selectFormat = {
+          // for selection drop down method
+          text: doc.data().name,
+          value: doc.data().name
+        };
+        teamNames.push(allNames); // push the data to the array
+        if (doc.data().status === '1') {
+          // private team
+          const temp: teamData = {
+            name: doc.data().name as string,
+            leader: doc.data().leader as string,
+            size: doc.data().members.length as number,
+            type: 'Private',
+            password: doc.data().password
+          };
+          teams.push(temp); // push the data to the array
+        } else {
+          // public team
+          const temp: teamData = {
+            name: doc.data().name as string,
+            leader: doc.data().leader as string,
+            size: doc.data().members.length as number,
+            type: 'Public',
+            password: doc.data().password
+          };
+          teams.push(temp); // push the data to the array
+        }
+      } else {
+        console.log(doc.data().name, ' is full'); // if the team is full
+      }
+    });
+    teams.sort((a: any, b: any) =>
+      a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1
+    );
+    setTeams(teams); // set the teams
+  }
+
+  // update the data when the page loads
+  // update the data when the teams are added, removed, or modified
+  useEffect(() => {
+    const unsubscribe = onSnapshot(
+      collection(FirestoreDB, 'teams'),
+      (collection: any) => {
+        const temp: any[] = [];
+        collection.forEach((doc: any) => {
+          temp.push(doc);
+        });
+        getData(temp);
+      }
+    );
+    return () => {
+      console.log('unsubscribing from team join page');
+      unsubscribe();
+    };
+  }, [ctx.user, ctx.team]);
+
+  // move to the team creation page
+  const moveToCreateTeam = () => {
+    history.push('/app/teamcreation');
+  };
+
+  // check team deadline
+  function displayPage() {
+    if (buttonValid === false) {
+      return <>{DisplayTeams(allTeams)}</>;
+    } else {
+      return (
+        <h1>The deadline to join or create a team was {adData.teamDate}.</h1>
+      );
+    }
+  }
+
+  return (
+    <IonPage>
+      <IonHeader>
+        <NavBar>
+          <IonTitle> Team Join </IonTitle>
+        </NavBar>
+      </IonHeader>
+      <IonContent fullscreen className="team-join">
+        <IonRow>
+          <IonCol sizeXs="12" sizeMd="6">
+            <IonCard className="join-create-team-card">
+              <IonCardHeader
+                style={{ display: 'flex', justifyContent: 'center' }}
+              >
+                <img alt="Picture of walktober team" src={team} />

```

Comment:
Can we please change this to something like "Art depicting 6 team members in a team huddle"?

Reply:
done.